tests/web3js/eth_deploy_contract_and_interact_test.js

```diff
@@ -173,4 +173,40 @@ it('deploy contract and interact', async() => {
         )
     }
 
+    // check that revert reason for custom error is correctly returned for gas estimation
```

Comment:
lets create maybe gast estimation test separately, because it feels to me we will have multiple tests with weird cases

Reply:
You are right. It has become quite large, I must admit. I'll leave it for another PR though.